src/DnsHelperUI/MainForm.cs

```diff
@@ -34,9 +34,23 @@ private void MainForm_Load(object sender, EventArgs e)
             // Load the DNS server config from the file
             ReloadContextMenuItems();
 
-            // Discover the current nic, just naively select the first one 
-            // (this will not work on systems with multiple enabled NIC's)
-            CurrentNic = NetworkManagement.GetAllNicDescriptions().FirstOrDefault();
+            // Get all the nics and load them into a drop down, 
+            // if we have any NICs, fill dropdown and 
+            // select the first one from the list as current.
+            var nics = NetworkManagement.GetAllNicDescriptions();
+            
+            if (nics.Count() > 0)
+            {
+                CurrentNic = nics.FirstOrDefault();
+                cmbNics.Items.AddRange(nics);
+            }
+            else
+            {
+                CurrentNic = "No available NiCs";
+                cmbNics.Items.Add("No available NiCs");
+            }
+
+            cmbNics.SelectedIndex = 0;
 
             // Update the DNS values to the currently applied ones
             var dnsServers = RefreshDNSValues();
```

Comment:
Delete this update sequence, redundant. See comment about `CurrentNic` property above